include/swift/Sema/ConstraintSystem.h

```diff
@@ -1632,6 +1632,13 @@ class Solution {
   /// A map from argument expressions to their applied property wrapper expressions.
   llvm::DenseMap<ASTNode, SmallVector<AppliedPropertyWrapper, 2>> appliedPropertyWrappers;
 
+  ArrayRef<AppliedPropertyWrapper> getAppliedPropertyWrappers(ASTNode anchor) {
+    auto found = appliedPropertyWrappers.find(anchor);
```

Comment:
IMO we really ought to be keying this with the callee locator itself, not just the anchor. For e.g key paths with subscript components you can have different callee locators that share the same anchor (although we don't currently support external property wrappers for subscripts)

Reply:
Yeah, we do know that argument/parameter pair this property wrapper corresponds to, should be able to use full locator while applying and re-create it while coercing arguments...